keen/keen-js
# Handle string results for "categorical" dataType

## What does this PR do?

Right now, `categorical` data can contain results that are either strings or numbers. This is a problem because string results can't be visualized in a pie, bar, or column chart in google charts.

To solve this problem, I created a new `dataType` in `parseRawData.js` called `cat-numeric`, which represents categorical data with numeric results, and left the `categorical` `dataType` to represent categorical data with string results. The "new" `categorical` `dataType` can only render a table (and JSON, in explorer), while `cat-numeric` can render bar, pie, column, and table charts. That way, users can't accidentally try to render a pie chart for string results.

## Where to start:

Most of the changes happen in `parseRawData` - it might be a good idea to figure out how that file works before looking at the changes.

## How to test this PR

* Create a `group_by` query in explorer that requires a target property. Set the target property to be any property that's a `string` type. (please see me for an example)
* Observe that the default view is a pie chart that doesn't render correctly
* Build keen.js in this branch, and reload the explorer with this version of keen.js.
* Rerun the query you created in step 1.
* The default view is now a table that renders correctly. The only visualization options for this query should be "table" or "json" in explorer.
